packages/@uppy/aws-s3-multipart/src/index.ts

```diff
@@ -752,6 +752,19 @@ export default class AwsS3Multipart<
         const etag = xhr.getResponseHeader('ETag')
         const location = xhr.getResponseHeader('Location')
 
+        // https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/getAllResponseHeaders#examples
+        const arr = xhr
+          .getAllResponseHeaders()
+          .trim()
+          .split(/[\r\n]+/)
+        const headersMap: Record<string, string> = {}
+        for (const line of arr) {
+          const parts = line.split(': ')
+          const header = parts.shift() as string
+          const value = parts.join(': ')
+          headersMap[header] = value
```

Comment:
```suggestion
          const separatorIndex = line.indexOf(': ')
          const header = line.slice(0, separatorIndex).toLowerCase()
          const value = line.slice(separatorIndex + 2)
          if (header in headersMap) {
           headersMap[header] += `, ${value}`
         } else {
           headersMap[header] = value
         }
```

Reply:
I like simple code, this is trying to be too smart imho at the cost of understand what's going on. I prefer to leave it as is.

Reply:
? I completely disagree, your version is the one trying to be smart, I find my version much more readable

Reply:
```suggestion
          const parts = line.split(': ')
          const header = parts.shift()!
          const value = parts.join(': ')
          headersMap[header] = value
```